api/entity_test.go

```diff
@@ -0,0 +1,297 @@
+		{"testdata/substance/substance-alcohol-additional.json"},
+		{"testdata/legal/legal-associations-activities-to-overthrow.json"},
+		{"testdata/legal/legal-associations-advocating.json"},
+		{"testdata/legal/legal-associations-engaged-in-terrorism.json"},
+		{"testdata/legal/legal-associations-membership-overthrow.json"},
+		{"testdata/legal/legal-associations-membership-violence-or-force.json"},
+		{"testdata/legal/legal-associations-terrorism-association.json"},
+		{"testdata/legal/legal-associations-terrorist-organization.json"},
+		{"testdata/legal/legal-court.json"},
+		{"testdata/legal/legal-investigations-debarred.json"},
+		{"testdata/legal/legal-investigations-history.json"},
+		{"testdata/legal/legal-investigations-revoked.json"},
+		{"testdata/legal/legal-police-additionaloffenses.json"},
+		{"testdata/legal/legal-police-domesticviolence.json"},
+		{"testdata/legal/legal-police-offenses.json"},
+		{"testdata/legal/legal-technology-manipulating.json"},
+		{"testdata/legal/legal-technology-unauthorized.json"},
+		{"testdata/legal/legal-technology-unlawful.json"},
+		{"testdata/psychological/psychological-competence.json"},
+		{"testdata/psychological/psychological-consultations.json"},
+		{"testdata/psychological/psychological-diagnoses.json"},
+		{"testdata/psychological/psychological-conditions.json"},
+		{"testdata/psychological/psychological-hospitalizations.json"},
+		{"testdata/package/comments.json"},
+		{"testdata/submission.json"},
+	}
+
+	for _, secTest := range tests {
+		section, err := ioutil.ReadFile(secTest.path)
+		if err != nil {
+			t.Logf("Received error: %v", err)
+			t.Fail()
+		}
+		// make a version of the content that will remain unaltered
+		unchangedLines := strings.Split(string(section), "\n")
+		// make a copy you can manipulate
+		alteredLines := make([]string, len(unchangedLines))
+
+		// Determine which lines should be changed
+		linesOfInterest := selectLinesString(unchangedLines)
+		if len(linesOfInterest) < 1 {
+			t.Logf("Found no mentions of 'type' in JSON")
+			t.Fail()
+		}
+
+		for targetIndex := range linesOfInterest {
+			copy(alteredLines, unchangedLines)
+			indexOfInterest := linesOfInterest[targetIndex]
+			alteredLines[indexOfInterest] = strings.Replace(alteredLines[indexOfInterest], alteredLines[indexOfInterest], `"type": "unsinn",`, 1)
+			replacedSection := strings.Join(alteredLines, "\n")
+
+			// Deserialize the initial payload from the replaced JSON
+			payload := &Payload{}
+			if err := payload.Unmarshal([]byte(replacedSection)); err != nil {
+				t.Fatalf("Failed to deserialize JSON: %v\n:Error: %v\n", replacedSection, err)
+			}
+
+			// Extract the entity interface of the payload, kicking off entity.Unmarshal()
+			_, err := payload.Entity()
+			if err == nil {
+				t.Logf("Should have received an error from payload.Entity, instead got nil for type %v", payload.Type)
+				t.Fail()
+			}
+		}
+	}
+}
+
+func selectLinesString(lines []string) []int {
```

Comment:
So, the purpose of this hack is to no change the type of anything that is in a collection? What happens when you do that? Does that mean that items that only exist inside of collections don't have their unmarshal error lines called?

Reply:
Yes that is what that means. So all of an entity's fields are called and trigger errors, but the items within the collections don't trigger errors. This is actually ok because we do trigger the collection type error.